libcrux-psq/benches/psq_v2.rs

```diff
@@ -0,0 +1,755 @@
+                    let _len_r = responder
+                        .write_message(registration_payload_responder, &mut msg_channel)
+                        .unwrap();
+
+                    let (_len_i_deserialized, _len_i_payload) = initiator
+                        .read_message(&msg_channel, &mut payload_buf_initiator)
+                        .unwrap();
+
+                    let initiator = initiator.into_transport_mode().unwrap();
+                    (initiator, msg_channel)
+                },
+                |(initiator, msg_channel)| {
+                    let _ = initiator
+                        .write_message(b"Transport Message", msg_channel)
+                        .unwrap();
+                },
+                BatchSize::SmallInput,
+            )
+        },
+    );
+
+    // Transport read message
+    c.bench_function(
+        &format!("[Registration] Transport Read {ciphersuite}"),
+        |b| {
+            b.iter_batched_ref(
+                || {
+                    let mut msg_channel = vec![0u8; 4096];
+                    let mut payload_buf_responder = vec![0u8; 4096];
+                    let mut payload_buf_initiator = vec![0u8; 4096];
+
+                    let mut initiator = registration_initiator::<PQ>(
+                        rand::rng(),
+                        ctx,
+                        aad_initiator_outer,
+                        aad_initiator_inner,
+                        &responder_ecdh_keys,
+                        &responder_pq_keys,
+                        &initiator_ecdh_keys,
+                    );
+
+                    let registration_payload_initiator = b"Registration_init";
+                    let _len_i = initiator
+                        .write_message(registration_payload_initiator, &mut msg_channel)
+                        .unwrap();
+
+                    let mut responder = build_responder::<PQ>(
+                        rand::rng(),
+                        ctx,
+                        aad_responder,
+                        &responder_ecdh_keys,
+                        &responder_pq_keys,
+                    );
+
+                    let (_len_r_deserialized, _len_r_payload) = responder
+                        .read_message(&msg_channel, &mut payload_buf_responder)
+                        .unwrap();
+
+                    let registration_payload_responder = b"Registration_respond";
+                    let _len_r = responder
+                        .write_message(registration_payload_responder, &mut msg_channel)
+                        .unwrap();
+
+                    let (_len_i_deserialized, _len_i_payload) = initiator
+                        .read_message(&msg_channel, &mut payload_buf_initiator)
+                        .unwrap();
+
+                    let mut initiator = initiator.into_transport_mode().unwrap();
+                    let _ = initiator
+                        .write_message(b"Transport Message", &mut msg_channel)
```

Comment:
This looks like it doesn't take any significant time (~359ns) for me. Shall we use a somewhat bigger payload to see a less noisy number? But we should also figure out what usual payload sizes are here in order to measure the right thing.